lib/Dialect/HW/HWOps.cpp

```diff
@@ -2855,6 +2863,11 @@ OpFoldResult ArrayGetOp::fold(FoldAdaptor adaptor) {
                                        intTy.getIntOrFloatBitWidth()));
   }
 
+  // array_get(array_inject(_, index, element), index) -> element
+  if (auto inject = getInput().getDefiningOp<ArrayInjectOp>())
+    if (getIndex() == inject.getIndex())
+      return inject.getElement();
+
```

Comment:
Do we have a test for this?

Reply:
Yes sir :saluting_face: https://github.com/llvm/circt/pull/8504/files#diff-5515faaa7267def62dc25befc3df06e49ea0af65cbc2cc34e75687998276a973R1085-R1091